torch/csrc/autograd/variable.cpp

```diff
@@ -154,6 +154,25 @@ AutogradMeta* materialize_autograd_meta(const at::TensorBase& self) {
   return get_autograd_meta(self);
 }
 
+void update_retains_grad(
```

Comment:
There are a couple assumption for this function that are worth mentioning I think:
- The AM is already populated
- The fn does exist (not AccumulateGrad here)
- The fn has no hooks already set